data/systemd/snapd.service.in

```diff
@@ -27,6 +27,8 @@ NotifyAccess=all
 SuccessExitStatus=42
 RestartPreventExitStatus=42
 KillMode=process
+#TODO: test with "shared"
+KeyringMode=inherit
```

Comment:
wouldn't every process we spawn how have access to the same keyring we inherited from systemd? I suppose with this we could simply call snapd-fde-keymgr without systemd-run too?

Reply:
* Yes, we will call everything else with access to the keyring. But we should really not call anything we cannot trust anyway as we run as root and anything we call could just dispatch anything to systemd. Let's say there is a vulnerability in sfdisk, we are already vulnerable then. Unless we start dropping privileges on whatever we execute.

* I can follow up with removing `systemd-run` from calling snapd-fde-keymgr. But it is a lot of changes in tests for very little.

Reply:
No, it's fine, I don't think there's an urgent need to do anything about snapd-fde-keymgr right now. I think we should look into ensuring that the external things we run are isolated from us in a reasonable manner, and preferably invoked through systemd-run where possible.